src/components/EventHandler.tsx

```diff
@@ -0,0 +1,65 @@
+
+import * as React from "react";
+import * as LibraryEntryPoint from "./LibraryEntryPoint";
+import { LibraryItem } from "./LibraryItem";
+
+/**
+ * The Event class stores the callback function together with a name
+ * that identifies it.
+ */
+export class Event {
+    name: string;
+    callback: Function;
+
+    constructor(name: string) {
+        this.name = name;
+        this.callback = null;
+    }
+
+    registerCallback(callback: Function) {
+        this.callback = callback;
+    }
+}
```

Comment:
This doesn't seem to be aligned with the referenced document. One `Event` has one `name`, and one or more `callback` functions. The `Reactor` is meant to contain a list of `Event`, each `Event` represents one event with a given string. For example if we have three callbacks for `itemClicked` and two callbacks for `itemExpanded`, then we will only have two `Event` inside `Reactor` (i.e. one for `itemClicked` and another for `itemExpanded`).